includes/class-dispatcher.php

```diff
@@ -169,11 +168,81 @@ public static function send_to_followers( $outbox_item_id, $batch_size = 50, $of
 		}
 	}
 
+	/**
+	 * Retry sending to followers.
+	 *
+	 * @param array $inboxes        The inboxes to retry.
+	 * @param int   $outbox_item_id The Outbox item ID.
+	 * @param int   $attempt        The attempt number.
+	 */
+	public static function retry_send_to_followers( $inboxes, $outbox_item_id, $attempt = 1 ) {
+		$retries = self::send_to_inboxes( $inboxes, $outbox_item_id );
+
+		// Retry failed inboxes.
+		if ( ++$attempt < 3 && ! empty( $retries ) ) {
+			self::schedule_retry( $retries, $outbox_item_id, $attempt );
+		}
+	}
+
+	/**
+	 * Send to inboxes.
+	 *
+	 * @param array $inboxes        The inboxes to notify.
+	 * @param int   $outbox_item_id The Outbox item ID.
+	 * @return array The failed inboxes.
+	 */
+	private static function send_to_inboxes( $inboxes, $outbox_item_id ) {
+		$json    = Outbox::get_activity( $outbox_item_id )->to_json();
+		$actor   = Outbox::get_actor( \get_post( $outbox_item_id ) );
+		$retries = array();
+
+		foreach ( $inboxes as $inbox ) {
+			$result = safe_remote_post( $inbox, $json, $actor->get__id() );
+
+			if ( is_wp_error( $result ) && in_array( $result->get_error_code(), self::$retry_error_codes, true ) ) {
+				$retries[] = $inbox;
+			}
+
+			/**
+			 * Fires after an Activity has been sent to an inbox.
+			 *
+			 * @param array  $result         The result of the remote post request.
+			 * @param string $inbox          The inbox URL.
+			 * @param string $json           The ActivityPub Activity JSON.
+			 * @param int    $actor_id       The actor ID.
+			 * @param int    $outbox_item_id The Outbox item ID.
+			 */
+			\do_action( 'activitypub_sent_to_inbox', $result, $inbox, $json, $actor->get__id(), $outbox_item_id );
+		}
+
+		return $retries;
+	}
+
+	/**
+	 * Schedule a retry.
+	 *
+	 * @param array $retries        The inboxes to retry.
+	 * @param int   $outbox_item_id The Outbox item ID.
+	 * @param int   $attempt        Optional. The attempt number. Default 1.
+	 */
+	private static function schedule_retry( $retries, $outbox_item_id, $attempt = 1 ) {
+		\wp_schedule_single_event(
+			\time() + ( $attempt * 5 * MINUTE_IN_SECONDS ),
```

Comment:
can we use something that increases the value exponentially? so that the last retry might be in an hour or later?

Reply:
Sure. It doesn't really make a huge difference with only two retries, fwiw. But we could change the minute to hours and retry after 5 and then another 10 hours

Reply:
Fixed in [f73e764](https://github.com/Automattic/wordpress-activitypub/pull/1336/commits/f73e76436e83a3edfd4c8c681cb740e157bee65e)